### PR TITLE
WT-7917 Add evergreen validation to s_all.

### PR DESCRIPTION
**Pull request by Siddhartha8899**

### CODE REVIEW

**Comment by Siddhartha8899** [comment]
The `if command -v evergreen ` checks if the evergreen binary is available on the machine. Mainly for two reasons:

- Evergreen is not installed on the Ubuntu machines that are spawned during the patch build.
- As Keith Smith mentioned, "we need to be careful that the suite still runs for stand-alone users who aren't building/testing in a MongoDB environment."

File: dist/s_all [hunk 1]
```diff
@@ -81,6 +81,9 @@ run "./s_clang-format"
 run "python prototypes.py"
 run "sh ./s_typedef -b"
 run "python test_tag.py"
+if command -v evergreen > /dev/null; then
+	run "evergreen --level notice validate ../test/evergreen.yml"
```
Comment (etienneptl) [inline dist/s_all:85]: What are the different levels ?

Comment (etienneptl) [reply dist/s_all:85]: As discussed:
```
--level value 
Specify lowest visible log level as string: 'emergency|alert|critical|error|warning|notice|info|debug|trace' (default: "info")
```

Comment (etienneptl) [reply dist/s_all:85]: With the `notice` level used, it does not print the current warning. We should leave it to the default value which works.

File: dist/s_all [hunk 2]
```diff
@@ -47,6 +47,12 @@ errchk()
 		return
 	fi
 
+	# Return if evergreen validate runs sucessfully.
+	if echo "$1" | grep -q evergreen && [ "$(cat $2)" = "../test/evergreen.yml is valid" ] ; then
```
Comment (etienneptl) [inline dist/s_all:51]: Is $2 a temporary file created?

**Review by etienneptl — APPROVED** [review]
LGTM, good work. I just have a couple of comments.

File: dist/s_all (hunk 2 above)
Comment (Siddhartha8899) [reply dist/s_all:51]: Yes, the $2 file is temporary. We create the file in `run()` and remove it in `errchk()`.

Comment (etienneptl) [inline dist/s_all:51]: Missing quotes around `$2`, see [SC2086]( https://github.com/koalaman/shellcheck/wiki/SC2086)

File: dist/s_all [hunk 3]
```diff
@@ -47,6 +47,12 @@ errchk()
...
+		rm -f $2
```
Comment (etienneptl) [inline dist/s_all:52]: Same, missing quotes

File: dist/s_all (hunk 2 above)
Comment (Siddhartha8899) [reply dist/s_all:51]: but isn't $2 a file name? (Just one word?)

Comment (etienneptl) [reply dist/s_all]: It's about preventing globbing and word splitting

**Review by etienneptl — APPROVED** [review]
Thanks for the last fixes !

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Pull request merged**